mods/page_save.js

```diff
@@ -154,7 +172,8 @@ PageSave.prototype.saveWikitext = function(restbase, req) {
         return restbase.post({
             uri: new URI([rp.domain, 'sys', 'action', 'edit']),
             headers: {
-                cookie: req.headers.cookie
+                cookie: req.headers.cookie,
+                'x-forwarded-for': req.headers['x-forwarded-for']
```

Comment:
I'm not sure that exposing internal IPs is actually a good idea. It might be a better idea to provide a fixed fall-back IP like 127.0.0.1 instead.